content/exercises/data-scenarios/02-vector-needs-reformatting/solution.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+# The data in a row looks like:
+#     {
+#         "x": -49.6612421393137,
+#         "y": 64.619373319576695,
+#         "cuteness_rating": 5.583,
+#         "blood_acetone_grams_per_ml": 4.34e-06,
+#         "blood_acetylcholine_grams_per_ml": 7.91e-08,
+#     }
+# ... but ogr2ogr doesn't know which fields are geometry fields.
+
+
+ogr2ogr -f "GeoJSON" \
+    -nln "kcbcc" \
+    -nlt "POINT" \
+    -dialect "sqlite" \
+    -sql "SELECT ST_GeomFromText('POINT(' || x || ' ' || y || ')', 4326) geom, cuteness_rating, blood_acetone_grams_per_ml, blood_acetylcholine_grams_per_ml from Sheet1" \
```

Comment:
This is one working solution, but may not be the best learning experience to introduce a SQL-based method of defining geometry from columns in an excel sheet. What alternatives (e.g. Python `pandas.read_excel` -> shapely geometry?) would be more useful to teach?

Reply:
Seems like this could also be possible via the creation of a `.vrt` file. E.g., https://gis.stackexchange.com/questions/327/how-can-i-convert-an-excel-file-with-x-y-columns-to-a-shapefile

Reply:
E.g., a `kcbcc_DS4_final_v2_final.vrt` like this:

```
<OGRVRTDataSource>
    <OGRVRTLayer name="Sheet1">
        <SrcDataSource>kcbcc_DS4_final_v2_final.xlsx</SrcDataSource>
        <SrcLayer>Sheet1</SrcLayer>
        <GeometryType>wkbPoint</GeometryType>
        <LayerSRS>WGS84</LayerSRS>
        <GeometryField encoding="PointFromColumns" x="x" y="y"/>
    </OGRVRTLayer>
</OGRVRTDataSource>
```

Then, with `ogr2ogr`:

```
ogr2ogr output.gpkg kcbcc_DS4_final_v2_final.vrt
```

Reply:
Nice, added! Do you know if there's a way to omit the coordinate columns from the attributes with the VRT method? I didn't see anything obvious in the docs.

Reply:
https://gdal.org/drivers/vector/vrt.html

> The optional reportSrcColumn attribute [of the GeometryField] can be used to specify whether the source geometry fields (the fields set in the field, x, y, z, m attributes) should be reported as fields of the VRT layer. It defaults to TRUE. If set to FALSE, the source geometry fields will only be used to build the geometry of the features of the VRT layer.

Editing the `GeometryField` in the VRT to look like this worked for me:

```
<GeometryField encoding="PointFromColumns" x="x" y="y" reportSrcColumn="FALSE" />
```

Reply:
You read much gooder than me :laughing:

Reply:
[6daf5eb](https://github.com/nsidc/qgreenland-2023-researcher-workshop/pull/5/commits/6daf5eb8c587ee13703a4ad059fd4c2b81b4a788)